### ISSUE TITLE
support for static linking of x11?

### ISSUE THREAD

**Issue by luxe** [glfw/glfw#1714]
we are building glfw from source in a different build system.  I was hoping to statically link against x11 libraries, but I'm seeing a lot of:  
```
_glfw_dlopen("libXxf86vm.so.1");
_glfw_dlopen("libXi-6.so");
...
```

Does this mean statically linking with x11 is not possible?

**Comment by elmindreda** [comment]
Yup, currently GLFW does not work with statically linked X11 libraries. Loading is done transparently to most of the code, though, so we could add a macro to disable this.

**Comment by luxe** [comment]
Nice, I'll try a macro then and see if we can get static linking going.

**Comment by elmindreda** [comment]
If you don't mind me asking, why are you linking the X libraries statically?

**Comment by luxe** [comment]
it seems like the default approach when using bazel's build system. It might turn out to be a bad idea, but I've had some success thus far in building/using x11 directly and also statically linking it with sdl2 and building apps that way.

the whole reason for doing this in the first place was that our build itself was not very reliable on different machines- we had to build inside a docker container or assume dependencies were already pre-installed, but wrong versions of dependencies could also cause issues.  So I thought I'd see how far I could get in rebuilding the world from source hermetically using hosted versions of the projects (such as glfw) (pulled in and built with bazel).

That way the same compiler is used, it points to the same sysroot and it builds all of those same dependencies that would normally be system pre-installed.  There are ways to build x11 as shard objects and ensure they are visible for glfw in bazel files as well, but its a little different and I haven't totally figured it out yet.

Hopefully the desire for static linking here makes sense

**Comment by alexfrasson** [comment]
Not sure how far you got with building glfw with bazel, but maybe [what I have gotten to work within my tiny project](https://github.com/alexfrasson/ChromaRenderer/blob/master/third-party/glfw/glfw.BUILD) could be useful for you guys.
@elmindreda it would be awesome if glfw would support bazel as well :wink:

**Comment by Vertexwahn** [comment]
FYI: Bazel support for glfw tested with Ubuntu, macOS, and Windows: https://github.com/bazelbuild/bazel-central-registry/pull/6541 - I can open PR to bring the `BUILD.bazel` and `MODULE.bazel` into this repo - not sure if build files form other build systems are accepted....
